packages/core/src/components/fields/MultiSchemaField.tsx

```diff
@@ -14,11 +14,16 @@ import {
 } from "@rjsf/utils";
 
 /** Type used for the state of the `AnyOfField` component */
-type AnyOfFieldState = {
+type AnyOfFieldState<S extends StrictRJSFSchema = RJSFSchema> = {
   /** The currently selected option */
   selectedOption: number;
+  realOptions: S[];
```

Comment:
Not super clear what `real` means and kind of implies that `options` are fake, so just rename this

```suggestion
  /* The option schemas after retrieving all $refs */
  retrievedOptions: S[];
```